### PR TITLE
VISI: searchopcodes is a viv cli function which allows more flexible... 

### PR DESCRIPTION
**Pull request by atlas0fd00m**
searching, either within a function or in the entire workspace for numbers or text or regex.

VISI: questions:
1) does this make anyone else happy? ;)
2) should this be wrapped into the standard "do_search()" cli function?  it feels like it should.

### CODE REVIEW

File: vivisect/cli.py [hunk 1]
```diff
@@ -217,6 +222,164 @@ def do_xrefs(self, line):
...
+                                    addthis = True
+
+                            else:
+                                if pattern in cmt:
+                                    addthis = True
+
+                    # search operands
+                    if options.searchOperands:
+                        defaultSearchAll = False
+                        for opidx in range(len(op.opers)):
```
Comment (rakuy0) [inline vivisect/cli.py]: `for opidx, oper in enumerate(op.opers):`

File: vivisect/cli.py [hunk 2]
```diff
@@ -217,6 +222,164 @@ def do_xrefs(self, line):
...
+
+                        else:
+                            if pattern in oprepr:
+                                addthis = True
+                    
+                    # only want one listing of each va, no matter how many times it matches
+                    if addthis:
+                        res.append(va)
+
+        except Exception, e:
```
Comment (rakuy0) [inline vivisect/cli.py]: e is unused, unless you meant to vprint it here?

**Comment by atlas0fd00m** [comment]
@rakuy0 i think this is much better.  check?
still waiting on input on whether this should all be a subset of "search", or is this a good name?  now is the time to change it.  @invisig0th is excellent at naming...  any word?

**Comment by rakuy0** [comment]
much better :+1:

**Comment by rakuy0** [comment]
@atlas0fd00m I don't see anything else on this PR to change (lack of unit tests is something we can figure out later). I'm cool with you merging this now. Your thoughts?

**Pull request merged**